[TensorExpr] Fast sigmoid for LLVM

zheng-xq opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#39717 [TensorExpr] Fast sigmoid for LLVM**

Differential Revision: [D21949849](https://our.internmc.facebook.com/intern/diff/D21949849)

zheng-xq committed:
[TensorExpr] Fast sigmoid for LLVM

[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/tensorexpr/llvm_codegen.cpp:
```diff
@@ -322,19 +322,24 @@ void LLVMCodeGenImpl::emitWrapper(const std::vector<llvm::Type*>& params) {
     }
     // TODO: fast exp
     // TODO: fast erf
     // TODO: fast sigmoid
```
Nit: remove the finished "TODO" please.

zheng-xq replied:
Done

ZolotukhinM commented on torch/csrc/jit/tensorexpr/codegen.h:
```diff
@@ -182,6 +186,11 @@ TORCH_API std::unique_ptr<CodeGen> CreateCodeGen(
     const std::vector<CodeGen::BufferArg>& params,
     at::Device device = at::kCPU);
 
+class TORCH_API GenericIntrinsicsExpander : public IRMutator {
```
Nit: we're creating an entire class (+ derived from it) just to redefine a single function. We could've get away with just a single free-standing function here.

zheng-xq replied:
This class and function is invoked implicity through accept_mutator. Unless we come up with more magic wrappers here, there is no easy way to call into this in our current infrastructure with a free standing function.

ZolotukhinM reviewed (changes requested):
Overall looks good, but please add some tests for this change.

zheng-xq committed:
Update on "[TensorExpr] Fast sigmoid for LLVM"


Differential Revision: [D21949849](https://our.internmc.facebook.com/intern/diff/D21949849)

[ghstack-poisoned]

ZolotukhinM reviewed (approved):
Looks good if the tests are there!
